documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -351,4 +353,258 @@ Control characters are not allowed in a specification.
+
+**How to fix the violation**: Ensure the `Resource` type model has the properties `name`, `type` and `id` and they are marked as `readOnly:true`. 
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2056" />R2056 RequiredReadOnlyProperties
+**Output Message**: Property '{0}' is a required property. It should not be marked as 'readonly'.
+
+**Description**: A model property cannot be both `readOnly` and `required`. A `readOnly` property is something that the server sets when returning the model object while required is a property required to be set when sending it as a part of the request body.
+
+**Why the rule is important**: SDK generation fails when this rule is violated.
+
+**How to fix the violation**: Ensure that the given property is either marked as `readonly: true` or `required` but not both.
+
+**Bad Example**:
+```
+"MyModel": {
+  "properties":{
+    "MyProp":{
+      "type": "string",
+      "description": "sample prop",
+      "readOnly": true
+    }
+  },
+  "required": ["MyProp"]
+}
+```
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2014" />R2014 SubscriptionIdParameterInOperations/ServiceDefinitionParameters
+**Output Message**: Parameter "subscriptionId" is not allowed in the operations section, define it in the global parameters section instead/Parameter "{0}" is referenced but not defined in the global parameters section of Service Definition
+
+**Description**: `subscriptionId` must not be an operation parameter and must be declared in the global parameters section.
+
+**Why the rule is important**: Per ARM guidelines, `subscriptionId` must be set as a property on the generated client instead of the method signature. 
+
+**How to fix the violation**: Remove `subscriptionId` from the operation parameters and add it to the global parameters section if it doesn't exist there.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2003" />R2003 ValidFormats
+**Output Message**: '{0}' is not a known format.
+
+**Description**: Only valid types are allowed for properties.
+
+**Why the rule is important**: Invalid formats can cause errors during code generation or result in erraneous generated code.
+
+**How to fix the violation**: Ensure format defined for property is valid. Please refer [here](http://swagger.io/specification/) for allowed types in OpenAPI.
```

Comment:
I believe autorest has a list of formats that may extend OpenAPI spec, you may want to point here https://github.com/Azure/autorest/blob/81d4d31d06637f4f9ef042d7f2ec64cfea29892f/docs/developer/validation-rules/valid-formats.md